Assets/_BuildingBlocks/Credentials/Scripts/KeyVault.cs

```diff
@@ -10,317 +10,212 @@
-    public enum AuthorizationType //todo: remove this enum
-    {
-        //Specific order in items used in dropdown index
-        Public = -1,
-        UsernamePassword = 0,
-        // InferableSingleKey = 1, //Single field key, token or code (we dont know specifically yet but can infer it)
-        Key,
-        BearerToken,
-        Code,
-        Token,
-        FailedOrUnsupported
-    }
-
     [CreateAssetMenu(fileName = "KeyVault", menuName = "ScriptableObjects/KeyVault", order = 1)]
     public class KeyVault : ScriptableObject
     {
         [TextArea(3, 10)] public string Description = "";
+        [SerializeField] private bool log = false;
+        [SerializeField] private Dictionary<Uri, StoredAuthorization.StoredAuthorization> storedAuthorizations = new();
+
+        private MonoBehaviour coroutineMonoBehaviour;
+        public UnityEvent<StoredAuthorization.StoredAuthorization> OnAuthorizationTypeDetermined = new();
+
+        public Dictionary<string, Type> expectedAuthorizationTypes = new()
+        {
+            { "tile.googleapis.com", typeof(Key) },
+            { "engine.tygron.com", typeof(Token) },
+            { "geo.tygron.com", typeof(Token) }
+        };
 
-        public List<KnownUrlAuthorizationType> knownUrlAuthorizationTypes = new()
+        private static readonly Dictionary<Type, Func<Uri, string[], StoredAuthorization.StoredAuthorization>> supportedAuthorizationTypes = new()
         {
-            // new KnownUrlAuthorizationType() { baseUrl = "https://tile.googleapis.com/v1/3dtiles/root.json", authorizationType = AuthorizationType.InferableSingleKey },
-            // new KnownUrlAuthorizationType() { baseUrl = "https://engine.tygron.com/web/3dtiles/tileset.json", authorizationType = AuthorizationType.InferableSingleKey },
-            // new KnownUrlAuthorizationType() { baseUrl = "https://api.pdok.nl/kadaster/3d-basisvoorziening/ogc/v1_0/collections/gebouwen/3dtiles/tileset.json", authorizationType = AuthorizationType.Public }
+            { typeof(Public), (uri, args) => new Public(uri) }, //start with public, to not add query parameters if not needed
+            { typeof(Key), (uri, args) => new Key(uri, args[0]) },
+            { typeof(Code), (uri, args) => new Code(uri, args[0]) },
+            { typeof(Token), (uri, args) => new Token(uri, args[0]) },
+            { typeof(UsernamePassword), (uri, args) => new UsernamePassword(uri, args[1], args[0]) }, //order is Password before Username, because we need a consistent order to use passwordOrKey
+            { typeof(BearerToken), (uri, args) => new BearerToken(uri, args[0]) }
         };
 
-        public Dictionary<Uri, StoredAuthorization.StoredAuthorization> storedAuthorizations = new();
+        private StoredAuthorization.StoredAuthorization CreateStoredAuthorization(Type storedAuthorizationType, Uri uri, params string[] args)
+        {
+            if (!storedAuthorizationType.IsSubclassOf(typeof(StoredAuthorization.StoredAuthorization)))
```

Comment:
Kan dit niet gewoon met `storedAuthorizationType is not StoredAuthorization.StoredAuthorization`?

Reply:
nee, want storedAuthorizationType is hier een `Type` en niet een StoredAuthorization